src/QMCDrivers/DMC/WalkerControl.cpp

```diff
@@ -205,26 +211,9 @@ void WalkerControl::branch(int iter, MCPopulation& pop, bool do_not_branch)
   // ranks sending walkers from other ranks have the lowest walker count now.
   untouched_walkers = std::min(untouched_walkers, walkers.size());
 
-  { // copy good walkers
+  {
     ScopedTimer copywalkers_timer(my_timers_[WC_copyWalkers]);
-    const size_t good_walkers = walkers.size();
```

Comment:
This is entirely manipulation and calls that should be private to MCPopulation so I refactored it into there.  This also allows unit testing this with MCPopulation which makes sense since code can be tested without the need for the WalkerControl code.